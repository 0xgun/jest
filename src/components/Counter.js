import React from "react";
import { useState } from "react";
const Counter = () => {
  const [Counter, setCounter] = useState(1);
  const incrementCount = () => {
    setCounter((prevCounter) => prevCounter + 1);
  };
  const decrementCount = () => {
    setCounter((prevCounter) => prevCounter - 1);
  };
  return (
    <div>
      <p data-testid="counter">Count:{Counter}</p>
      <button data-testid="increment" onClick={incrementCount}>
        +
      </button>
      <button data-testid="decrement" onClick={decrementCount}>
        -
      </button>
    </div>
  );
};

export default Counter;
