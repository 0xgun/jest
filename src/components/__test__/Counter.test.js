import { render, fireEvent, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Counter from "../Counter";
describe("Counter", () => {
  test("Should increment Counter", () => {
    //render component in the virtual dom
    render(<Counter />);
    //select the elements to interact
    const counter = screen.getByTestId("counter");
    const increment = screen.getByTestId("increment");
    //iteract with elements
    fireEvent.click(increment);
    //assertions
    expect(counter).toHaveTextContent("2");
  });
  test("Should decrement Counter", () => {
    render(<Counter />);
    const counter = screen.getByTestId("counter");
    const decrement = screen.getByTestId("decrement");
    fireEvent.click(decrement);
    expect(counter).toHaveTextContent("0");
  });
});

describe("Counter Snapshot", () => {
  it("should matches DOm shapshot", () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
