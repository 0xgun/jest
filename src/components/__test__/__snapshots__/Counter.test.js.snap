// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot should matches DOm shapshot 1`] = `
<div>
  <p
    data-testid="counter"
  >
    Count:
    1
  </p>
  <button
    data-testid="increment"
    onClick={[Function]}
  >
    +
  </button>
  <button
    data-testid="decrement"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
